src/PIL/ImageFont.py

```diff
@@ -213,6 +214,8 @@ def load_from_bytes(f):
             )
 
         if is_path(font):
```

Comment:
I'm looking at `_util.is_path`, it checks `isinstance(f, (bytes, str, Path))`, so bytes are also handled as paths, but next I see: `else: load_from_bytes(font)`.

Do we really want consider bytes as paths? I believe this is obsolete in Python3

Reply:
#258 was the original request for this functionality.

Reply:
I assume `load_from_bytes` is unreachable here, so we need to remove it, or provide a way to reach it

Reply:
I think it's reachable. The logic hasn't changed since `load_from_bytes()` was added in #3785.

If it is a path, then pass the path to the C layer. If we're on Windows and the path is not ASCII, then open a file pointer to the path and read the contents of the file pointer with `load_from_bytes()`.
If it is not a path, then it is a file pointer, read the contents of the file pointer with `load_from_bytes()`.

Reply:
Ok, I see. I believe there is misnaming here, since f is not a bytes, it's file object.

Reply:
Yep, I've pushed a commit. I presume from the naming that the type hint also includes `io.BytesIO`

Reply:
@radarhere I see you've used `typing.IO`. This causes a mypy error, see #7642.

Testing with both PyCharm and mypy, I find that a `io.BytesIO` value is accepted for a `typing.BinaryIO` type hint, so I'm not sure I understand your last message.

Reply:
I think I was concerned about strings being used, but looking now, that doesn't appear to be a problem.